tests/network/wireguard.pm

```diff
@@ -63,19 +76,61 @@ sub run {
     barrier_wait 'VPN_ESTABLISHED';
     assert_script_run "ping -c10 $vpn_remote";
 
-    zypper_call 'in iperf';
-
     if (get_var('IS_MM_SERVER')) {
         assert_script_run "iperf3 --bind $vpn_local --server --daemon --port 5001";
         script_retry 'ss -lptn | grep 5001', delay => 3, retry => 3;
         mutex_create 'server_ready';
-        wait_for_children;
     } else {
         mutex_unlock 'server_ready';
         script_retry "iperf3 --bind $vpn_local --time 30 --client $vpn_remote --port 5001", timeout => 60, delay => 3, retry => 3;
     }
+    barrier_wait('IPERF_COMPLETED');
 
     assert_script_run 'ip link set down dev wg0';
+    assert_script_run 'ip link delete dev wg0';
+
+    ## Test wg-quick
+    assert_script_run('cd /etc/wireguard');
+    if (get_var('IS_MM_SERVER')) {
+        # Prepare new keys
+        assert_script_run('wg genkey | tee server | wg pubkey > server.pub');
```

Comment:
Well, as we do not use `$PIPESTATUS` if `wg genkey` fails we don't get it.

Reply:
It might be better to use mdoucha's shell runner (https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/9219) instead of writing shell in perl.

Reply:
```suggestion
        assert_script_run('wg genkey | tee server');
        assert_script_run('wg pubkey < server | tee server.pub');
```

Reply:
True! I've added `set -o pipefail` to account for failing commands in a pipe.

I agree that it would be a good case for using the shell runner. However I was extending an existing test case and didn't had the motivation to migrate everything from perl to shell - It would be a good consideration for the future though.